test/demo/raiseIntent.test.ts

```diff
@@ -0,0 +1,203 @@
+                            testAppIdentity.appId
+                        );
+
+                        // App should now be running
+                        await expect(fin.Application.wrapSync(testAppIdentity).isRunning()).resolves.toBe(true);
+
+                        const listener = await fdc3Remote.getRemoteIntentListener(testAppIdentity, validPayloadPreregistered.intent);
+                        const receivedContexts = await listener.getReceivedContexts();
+
+                        expect(receivedContexts).toEqual([validPayloadPreregistered.context]);
+
+                        await fin.Application.wrapSync(testAppIdentity).quit();
+                    });
+                });
+
+                describe('When the target is *not* registered to accept the raised intent', () => {
+                    // TBD: should the app open if the intent is not valid?
+                    test.todo('When calling raiseIntent [behaviour TBD]');
+                });
+                describe('When the target is not in the directory', () => {
+                    // Currently this will just open the resolver UI, which is probably the wrong behaviour
+                    // TODO: Make the service return an error when targeting an app which isn't in the directory
+                    test.skip('When calling raiseIntent the promise rejects with an error', async () => {
+                        const resultPromise = fdc3Remote.raiseIntent(testManagerIdentity, validPayload.intent, validPayload.context, 'this-app-does-not-exist');
+                        // TODO: decide what the error should be
+                        await expect(resultPromise).rejects.toThrow();
+                    });
+                });
+            });
+
+            describe('And the listener is added after a delay', () => {
+                const testAppIdentity = {
+                    uuid: 'test-app-1',
+                    name: 'test-app-1',
+                    appId: '550'
+                };
+                test('The targeted app opens and its listener is triggered exactly once with the correct context, after the listener is set up', async () => {
+                    // We dont await for this promise here because it wouldn't resolve.
+                    // It's going to resolve only after we add the listener to the test app
+                    const raiseIntentPromise = fdc3Remote.raiseIntent(
+                        testManagerIdentity,
+                        validPayload.intent,
+                        validPayload.context,
+                        testAppIdentity.appId
+                    );
+
+                    await delay(1500);
+                    // App should now be running
+                    await expect(fin.Application.wrapSync(testAppIdentity).isRunning()).resolves.toBe(true);
+
+                    // We want to have a delay between the app running and the intent listener being set up
```

Comment:
This comment would be more helpful if it said why we want to wait. What are we waiting for here?

Reply:
done